lib/src/domain/use_cases/implementation/favorites_use_case.dart

```diff
@@ -0,0 +1,26 @@
+import '../../../core/utils/resources/data_state.dart';
+import '../../../data/repositories/database_repository.dart';
+import '../../entities/favorite_entity.dart';
+import '../i_use_case.dart';
+
+class FavoritesUseCase implements IUseCase {
+  final DatabaseRepository databaseRepository;
+
+  FavoritesUseCase({required this.databaseRepository});
+
+  @override
+  Future<DataState<List<FavoriteEntity>>> call({params}) async {
+    if (params != null) {
+      var favorite = FavoriteEntity(id: params);
+      await databaseRepository.updateFavorite(favorite);
+    }
```

Comment:
I'd split this into two separate usecases, because I needed to read the code to understand what this UseCase does.